static/js/editor.ts

```diff
@@ -1,109 +1,88 @@
 import { Markers } from "./markers";
 
 export type Breakpoints = Record<number, string>;
+export enum EditorType {
+  MAIN,
+  MODAL,
+  COMMON_MISTAKES,
+  CHEATSHEET,
+  PARSONS,
+  EXAMPLE
```

Comment:
All of them require different options! And I didn't want to expose a `setOptions` method, so I decided to set that up in the constructors. Since there are different kinds of these blocks I created this enum to configure all of them in the constructor. You can see that in the constructor.